routes/lists.js

```diff
@@ -23,65 +24,73 @@ router
       throw new HttpError(422, `${id} is not a valid ObjectId`);
     }
 
-    User.findById(userId)
-      .then(user => {
-        const list = user.shoppingLists.id(id);
-        if (!list) {
+    ShoppingList.findById(id)
+      .populate('store')
+      .then(shoppingList => {
+        if (!shoppingList) {
+          throw new NotFoundError();
+        }
+
+        // TODO: see if there is a better way of comparing ObjectIds
```

Comment:
apparently this can be done with id.equals(mongoose.types.Type.id(id)) (something like this)

Reply:
Alternatively, we could have done `ShoppingList.findOne({_id: id, user: userId})`